pkg/jsonmessage/jsonmessage.go

```diff
@@ -21,6 +22,10 @@ func (e *JSONError) Error() string {
 	return e.Message
 }
 
+// JSONProgress wraps a Progress
+// terminalFd is the fd of the current terminal
+// Total is the total of the progress, Current is the
+// current of the progress, start is the Start of the progress.
```

Comment:
It's good that you have these descriptions, but please move them into the struct so that godoc will pick them up. Also it would be good to go into more detail here, like what units are used.